### PR TITLE
Add basic implementation of greeting method to FE2

### PR DESCRIPTION
**Pull request by ahominal**
This pull request follows Nico's suggestion of implementation of the new the protocol for encrypted e-voting described here https://github.com/dedis/popstellar/pull/917 .

This pull request contains:
- basic "Greeting" message implementation
- basic "Server / Peer object" saving backend's info on frontend
- handling of the Greeting
- added global Server Repository (Map<LaoId, Server>)

### CODE REVIEW

**Review by GabrielFleischer — CHANGES_REQUESTED** [review]
Please merge master into the branch 😉

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/ServerRepository.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    serverByLaoId.put(laoId, server);
+  }
+
+  /**
+   * Get the corresponding server to the given Lao Id (if present)
+   */
+  public Server getServerByLaoId(String laoId){
+    if (serverByLaoId.containsKey(laoId)) {
+      return serverByLaoId.get(laoId);
+    } else {
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/ServerRepository.java:39]: You can drop the else here

**Review by Tesa27 — CHANGES_REQUESTED** [review]
Looks good overall, I did not go into detail on this PR but seems to be coherent with fe1, just a minor detail.

File: fe2-android/app/src/test/java/com/github/dedis/popstellar/utility/handler/ChirpHandlerTest.java [hunk 2]
```diff
@@ -40,6 +43,7 @@
 
 import io.reactivex.Completable;
 
+@HiltAndroidTest
```
Comment (GabrielFleischer) [inline fe2-android/app/src/test/java/com/github/dedis/popstellar/utility/handler/ChirpHandlerTest.java:46]: You do not need to use Hilt there. You can just create the serverRepository through its constructor.
`new ServerRepository()`

File: fe2-android/app/src/test/java/com/github/dedis/popstellar/utility/handler/ConsensusHandlerTest.java [hunk 3]
```diff
@@ -60,6 +63,7 @@
 
 import io.reactivex.Completable;
 
+@HiltAndroidTest
```
Comment (GabrielFleischer) [inline fe2-android/app/src/test/java/com/github/dedis/popstellar/utility/handler/ConsensusHandlerTest.java:66]: Same here

File: fe2-android/app/src/test/java/com/github/dedis/popstellar/utility/handler/RollCallHandlerTest.java [hunk 4]
```diff
@@ -49,6 +52,7 @@
 
 import io.reactivex.Completable;
 
+@HiltAndroidTest
```
Comment (GabrielFleischer) [inline fe2-android/app/src/test/java/com/github/dedis/popstellar/utility/handler/RollCallHandlerTest.java:55]: And here 😄

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/lao/GreetLao.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+public class GreetLao extends Data {
+
+  @NonNull
+  @SerializedName("lao")
+  private final String id;
+
+  // Backend sender address
+  @NonNull
+  @SerializedName("frontend")
+  private String frontendKey;
```
[the remark this answers is not in the document]
Comment (ahominal) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/lao/GreetLao.java:23]: Ohh that's why i didn't use it, thanks !

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/ServerRepository.java (hunk 1 above)
Comment (ahominal) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/ServerRepository.java:39]: thanks for noticing !

**Review by GabrielFleischer — APPROVED** [review]
LGTM

**Review by zoemarin — APPROVED** [review]
Seems Fine to me :)

**Review by Tesa27 — APPROVED** [review]
LGTM but for some reason tests for fe1 keep failing

**Comment by Tyratox** [comment]
Yes the tests for fe-1 fail because the protocol specification (json schema) was changed but this PR does not contain the changes done to fe-1 to support these changes. This is fine since I will merge https://github.com/dedis/popstellar/pull/957 to the same branch as this PR and then both tests should pass :)

**Pull request merged**